src/scheduler.cpp

```diff
@@ -113,11 +123,9 @@ void Scheduler::unbind() {
 }
 
 Scheduler::Scheduler(const Config& config)
-    : cfg(config), workerThreads{}, singleThreadedWorkers(config.allocator) {
-  if (cfg.workerThread.count > 0 && !cfg.workerThread.affinityPolicy) {
-    cfg.workerThread.affinityPolicy = Thread::Affinity::Policy::anyOf(
-        Thread::Affinity::all(cfg.allocator), cfg.allocator);
-  }
+    : cfg(setConfigDefaults(config)),
+      workerThreads{},
+      singleThreadedWorkers(config.allocator) {
```

Comment:
Not a major thing, but if ever setConfigDefaults did something with the allocator member, you'd probably want to init singleThreadedWorkers with `cfg.allocator` here.

Reply:
Ack - thanks for pointing this out. I did spot this, but I'm really not keen on requiring inter-field initialization ordering.
Leaving as is for now.